web/test/pages/dashboard.test.tsx

```diff
@@ -668,33 +668,173 @@ describe("dashboard page", () => {
+          });
+          renderDashboardPage({
+            quickActionLinks: [
+              generateQuickActionLink({
+                name: "some name",
+                filename: "state-contracting-external-link",
+                externalRoute: "some-external-route",
+              }),
+            ],
+          });
+          expect(screen.queryByText("some name")).not.toBeInTheDocument();
+        }
+      );
+
+      it.each(operatingPhasesWithQuickActions)(
+        "displays state-contracting-external-link quick action when business persona is foreign, industry id is not HIC and sector id is not construction for %s",
+        (phase) => {
+          useMockProfileData({
+            operatingPhase: phase,
+            sectorId: "some-value",
+            businessPersona: "FOREIGN",
+            industryId: "generic",
+          });
+          renderDashboardPage({
+            quickActionLinks: [
+              generateQuickActionLink({
+                name: "some name",
+                filename: "state-contracting-external-link",
+                externalRoute: "some-external-route",
+              }),
+            ],
+          });
+          expect(screen.getByText("some name")).toBeInTheDocument();
+        }
+      );
+
+      it.each(operatingPhasesWithQuickActions)(
+        "does not display state-contracting-external-link quick action when business persona is foreign, industry id is HIC and sector id is not construction for %s",
+        (phase) => {
+          useMockProfileData({
+            operatingPhase: phase,
+            sectorId: "some-value",
+            businessPersona: "FOREIGN",
+            industryId: "home-contractor",
+          });
+          renderDashboardPage({
+            quickActionLinks: [
+              generateQuickActionLink({
+                name: "some name",
+                filename: "state-contracting-external-link",
+                externalRoute: "some-external-route",
+              }),
+            ],
+          });
+          expect(screen.queryByText("some name")).not.toBeInTheDocument();
+        }
+      );
+
+      it.each(operatingPhasesWithQuickActions)(
+        "does not display state-contracting-external-link quick action when business persona is foreign, industry id is not HIC and sector id is construction for %s",
+        (phase) => {
+          useMockProfileData({
+            operatingPhase: phase,
+            sectorId: "construction",
+            businessPersona: "FOREIGN",
+            industryId: "generic",
+          });
+          renderDashboardPage({
+            quickActionLinks: [
+              generateQuickActionLink({
+                name: "some name",
+                filename: "state-contracting-external-link",
+                externalRoute: "some-external-route",
+              }),
+            ],
+          });
+          expect(screen.queryByText("some name")).not.toBeInTheDocument();
+        }
+      );
+    });
```

Comment:
Aren't you missing 2 cases here for a foreign and poppy when it is both industry id is HIC and sector id is construction?

You have 
not HIC and not construction
not HIC and is construction
is HIC and not construction

If I'm not mistaken (I could be its, late)